### ISSUE TITLE
Audit and refactor Material components with Material React components

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#2510]
## Feature Description

While reviewing #2189 I noticed we have many generic components which seem to reimplement existing `@material/react-*` components. Perhaps some of these components didn't exist at the time or there may be reasons why we needed to use alternate versions instead, but this should be re-evaluated to make sure we aren't "reinventing the wheel" unnecessarily.

We currently only pull in 3 Material React packages
https://github.com/google/site-kit-wp/blob/05a4265cf21c4ed9a10d8623e756c953fd7b2bb8/package.json#L109-L111

However, we use several other material components that also have Material React packages
https://github.com/google/site-kit-wp/blob/05a4265cf21c4ed9a10d8623e756c953fd7b2bb8/package.json#L100-L108
https://github.com/google/site-kit-wp/blob/05a4265cf21c4ed9a10d8623e756c953fd7b2bb8/package.json#L112-L115

* https://www.npmjs.com/package/@material/react-button
* https://www.npmjs.com/package/@material/react-checkbox
* https://www.npmjs.com/package/@material/react-dialog
* https://www.npmjs.com/package/@material/react-linear-progress
* etc

We import and re-export our material components in [this index module](https://github.com/google/site-kit-wp/blob/05a4265cf21c4ed9a10d8623e756c953fd7b2bb8/assets/js/material-components/index.js) although this mixes React components (e.g. `TextField`) with Material web component classes (underlying component abstraction) e.g. `MDCFormField`.

We shouldn't need to use `MDC*` classes directly as these should be implemented in the `react-*` components for us.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* All material components should be sourced from `@material/react-*` packages where a custom implementation is not needed
	- All custom implementations (if needed) must be documented accordingly in the component module why `@material/react-*` cannot be used
* `MDC*` component _classes_ should not be used directly
* `@material/*` packages that do not start with `react-` should be removed unless explicitly needed by a custom implementation that cannot be replaced by a `@material/react-*` package
    - These will likely remain installed but as dependencies of their respective `@material/react-*` packages
* `@material/theme` should be removed if not needed (doesn't seem to be used anywhere)
* `assets/js/material-components/index.js` should be removed and components should be imported from the packages directly as "External dependencies"
* No visual or behavioral changes should happen as a result, although minor differences may be acceptable if the result is enhanced or otherwise preferable (such as avoiding the need to re-implement a component)

## Implementation Brief

* 

### Test Coverage

* 

### Visual Regression Changes

* 

## QA Brief

* 

## Changelog entry

*

**Comment by aaemnnosttv** [comment]
cc: @felixarntz

**Comment by felixarntz** [comment]
@aaemnnosttv I agree we should do this, but I think we can defer that work to sometime next year. It's a lower priority task that will likely involve some work. Let's wait until most/all of the JS refactoring epic is done. (While this is also somewhat refactoring, I think it's decoupled from the current epic.)

**Comment by felixarntz** [comment]
@aaemnnosttv This has been stalled for a while, will close this now in favor of #5801.
